### PR TITLE
Add type annotation on `concurrency/threads.py`

### PR DESCRIPTION
**Pull request by Kludex**
- Related to #7394

### CODE REVIEW

File: celery/concurrency/thread.py [hunk 1]
```diff
@@ -1,45 +1,67 @@
...
                                  callback, accept_callback)
         return ApplyResult(f)
 
-    def _get_info(self):
+    def _get_info(self) -> PoolInfo:
         return {
             'max-concurrency': self.limit,
             'threads': len(self.executor._threads)
-            # TODO use a public api to retrieve the current number of threads
-            # in the executor when available. (Currently not available).
```
Comment (Kludex) [inline celery/concurrency/thread.py:44]: This will never happen.

File: celery/concurrency/thread.py [hunk 2]
```diff
@@ -1,45 +1,67 @@
...
+    def on_stop(self) -> None:
         self.executor.shutdown()
         super().on_stop()
 
-    def on_apply(self, target, args=None, kwargs=None, callback=None,
-                 accept_callback=None, **_):
+    def on_apply(
+        self,
+        target: TargetFunction,
+        args: tuple | None = None,
```
Comment (Kludex) [inline celery/concurrency/thread.py:53]:
```suggestion
        args: tuple[Any, ...] | None = None,
```

File: celery/concurrency/thread.py [hunk 3]
```diff
@@ -1,45 +1,67 @@
...
 
-from concurrent.futures import ThreadPoolExecutor, wait
+from concurrent.futures import Future, ThreadPoolExecutor, wait
+from typing import TYPE_CHECKING, Any, Callable
 
 from .base import BasePool, apply_target
 
 __all__ = ('TaskPool',)
 
+if TYPE_CHECKING:
```
Comment (Kludex) [inline celery/concurrency/thread.py:11]: Everything inside this block should have been ignored by codecov, according to https://github.com/celery/celery/blob/651095e3602756237920f4fa7ac170e1322c1939/pyproject.toml#L25

Why isn't it?

Comment (auvipy) [reply celery/concurrency/thread.py:11]: the config is of pre mypy era of celery, so we can adjust that

Comment (Kludex) [reply celery/concurrency/thread.py:11]: I don't know what needs to be adjusted on codecov. :eyes:

Comment (auvipy) [reply celery/concurrency/thread.py:11]: btw codecov is buggy

Comment (Kludex) [reply celery/concurrency/thread.py:11]: Do you want me to open a PR removing it, and calculate the coverage exclusively with `coverage`?

Comment (auvipy) [reply celery/concurrency/thread.py:11]: I saw some projects moved to that, you can come with that

**Comment by Kludex** [comment]
@auvipy anything else that I can do for this PR? I can clarify anything needed here.

**Pull request merged**